lib/rules/selector-class-pattern/index.mjs

```diff
@@ -123,32 +123,31 @@ const rule = (primary, secondaryOptions) => {
  */
 function hasInterpolatingAmpersand(selector) {
 	for (const [i, char] of Array.from(selector).entries()) {
-		if (char !== '&') {
-			continue;
-		}
-
-		const prevChar = selector.charAt(i - 1);
-
-		if (prevChar && !isCombinator(prevChar)) {
-			return true;
-		}
+		if (char !== '&') continue;
 
-		const nextChar = selector.charAt(i + 1);
-
-		if (nextChar && !isCombinator(nextChar)) {
-			return true;
-		}
+		if (!hasAdjacentCombinator(selector, i)) return true;
 	}
 
 	return false;
 }
 
 /**
- * @param {string} x
- * @returns {boolean}
+ * @param {string} selector
+ * @param {number} index
  */
-function isCombinator(x) {
-	return /[\s+>~]/.test(x);
+function hasAdjacentCombinator(selector, index) {
+	const prevChar = selector.charAt(index - 1);
+	const fn = (/** @type {string} */ character) => /[\s+>~]/.test(character);
+
+	if (prevChar && fn(prevChar)) return true;
+
+	if (prevChar === '|' && selector.charAt(index - 2) === '|') return true;
```

Comment:
note

`charAt` behaves differently with negative numbers.
i.e. using `at` would be unsafe